appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java

```diff
@@ -617,23 +550,92 @@
   @Description("Name of My Projects tab")
   String myProjectsTabName();
 
-  @DefaultMessage("Guide")
-  @Description("Name of Guide link")
-  String guideTabName();
-
-  @DefaultMessage("Report an Issue")
-  @Description("Link for Report an Issue form")
-  String feedbackTabName();
-
-  @DefaultMessage("Gallery")
+  @DefaultMessage("App Gallery")
   @Description("Link for Gallery")
   String galleryTabName();
 
+  @DefaultMessage("Teach")
+  @Description("Link for Educator website")
+  String teachTabName();
+
+  @DefaultMessage("Moderation")
+  @Description("Name of moderation tab")
+  String moderationTabName();
+
+  //Gallery
+  @DefaultMessage("Profile")
+  @Description("Label of the link for private user profile")
+  String privateProfileLink();
+
+  //About
+  @DefaultMessage("About")
+  @Description("Label of the button leading to About related cascade items")
+  String aboutTabName();
+
+  @DefaultMessage("App Inventor Version")
+  @Description("Label of the link for About")
+  String appInventorVersionMenuItem();
+
+  @DefaultMessage("Current Companion Version")
+  @Description("Information about the Companion")
+  String companionInformation();
+
+  @DefaultMessage("Update the Companion")
+  @Description("Menu item to update the Companion to the latest version")
+  String companionUpdate();
+
+  @DefaultMessage("You must have a project open to update the Companion")
+  @Description("")
+  String companionUpdateMustHaveProject();
+
+  @DefaultMessage("Update Companion on Emulator")
+  @Description("Hard Reset the Emulator.")
+  String hardResetConnectionsMenuItem();
```

Comment:
@ajcolter I'm not sure I'm happy with this wording. Hard Reset does *not* update the Companion on the Emulator. Instead it resets it to its initial as installed state (which at this point is an ancient companion). The auto-update mechanism then will prompt to update it on next connection. Maybe we should just remove this item now that we have the "Update the Companion" functionality. (I'll double check that it works on the emulator first, but it should).